Makefile

```diff
@@ -145,6 +145,13 @@ TF_CFLAGS_aarch32	=	$(march32-directive)
 TF_CFLAGS_aarch64	=	-march=armv8-a
 endif
 
+# Check if ld.bfd exists directly (if it's an absolute path) or is in PATH (if
+# it's just a basename). whereis needs trailing dot because it cuts extensions.
+ifeq ($(shell test -x ${LD}.bfd || whereis ${LD}.bfd. | grep -q / ; echo $$?),0)
```

Comment:
We try to avoid direct calls into the shell for compatibility with Windows environments. There's an abstraction layer in build_env.mk to help with this sort of thing.

Reply:
Oh, okay. It doesn't offer the kind of primitives I need for this, unfortunately. And I think whereis doesn't even exist on Windows.

But gold doesn't support non-ELF platforms anyway, so this patch isn't necessary for any of the non-Unix environments. Would it be okay if I just guard this code with ifdef UNIX_MK?

Reply:
I'd prefer this check to be abstracted into a new command in build_env.mk, with this implementation moved to unix.mk, and the windows.mk version always returning false.

Reply:
Actually, I figured out that it's not that hard to just reimplement the functionality of whereis within make, since we have the $PATH variable available from the environment. I'll do that and then it doesn't need any shell commands.